### PR TITLE
Bump WordPress and PHP minimum

### LINKED ISSUE #905: Bump WordPress and PHP minimums
Bump WP & PHP Minimums to the following
 
- [x] WordPress to 5.7
- [x] PHP to 7.4
- [x] Remove any no-longer-needed conditional code for older version support.
- [x] Take a look at adjusting our testing matrices, if applicable.
- [x] Bump the minimums in plugin documentation and header fields.

Misc
- [x] remove references to gutenberg_get_jed_locale_data
- [x] Application passwords always available (even if they are not enabled)

### LINKED ISSUE #925: Remove application passwords dependency
### Is your enhancement related to a problem? Please describe.

10up install the application passwords plugin, which is abandoned and the feature is included with WP core

### Designs

_No response_

### Describe alternatives you've considered

> Package georgestephanis/application-passwords is abandoned, you should avoid using it. No replacement was suggested.

This feature is now in core WordPress, no need to install the plugin.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by peterwilsoncc**
### Description of the Change

Bump WordPress minimum to 5.7
Bump PHP minimum to 7.4

Closes #905, Closes #925

### Changelog Entry

> Changed - Now requires PHP 7.4 or later and WordPress 5.7 or later.

### Credits

Props @peterwilsoncc, @vikrampm1, @iamdharmesh.

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: distributor.php [hunk 1]
```diff
@@ -55,14 +55,29 @@ function( $class ) {
 );
 
 /**
- * Require PHP version 5.6 - throw an error if the plugin is activated on an older version.
+ * Require PHP 7.4+, WP 5.7+ - throw an error if the plugin is activated on an older version.
  */
 register_activation_hook(
```
Comment (peterwilsoncc) [inline distributor.php:93]: These changes prevent activation of the plugin on unsupported versions of WP, PHP.

There is now the small risk a site will have an active plugin and update to a version it no longer supports. How have we handled that in the past?

Comment (dkotter) [reply distributor.php:93]: So I was trying to find a version of this in our plugins and couldn't quite remember where it's at but what we've done in the past is have some code in the main plugin file (so this file in this case) that checks if our requirements are met and if not, output an admin notice and stop execution of the plugin. Not our plugin but here's a similar approach in a Woo plugin: https://github.com/woocommerce/google-listings-and-ads/blob/develop/google-listings-and-ads.php#L41 and https://github.com/woocommerce/google-listings-and-ads/blob/develop/src/Internal/Requirements/VersionValidator.php#L22

Comment (peterwilsoncc) [reply distributor.php:93]: In 6187c5625d1f612a7715b3be0c9da9bf0c725c7a I've added a first pass at showing a notice if the plugin fails to meet minimum version requirements.

Honestly, it seems like a lot for something that ought to be quite minor but I am not sure if there is much that can be done about that. For this to work, the plugin's main file will always need to be compatible with earlier versions of WP/PHP. The up-to-date code will need to come later.

Should I move some of the bootstrapping code so `distributor.php` can focus on activation & a bootstrapper can focus on bootstrapping?

File: includes/syndicated-post-ui.php [hunk 2]
```diff
@@ -26,12 +28,8 @@ function() {
...
 			add_filter( 'post_row_actions', __NAMESPACE__ . '\remove_quick_edit', 10, 2 );
 
-			$post = isset( $_GET['post'] ) ? get_post( (int) $_GET['post'] ) : false; // @codingStandardsIgnoreLine Nonce not required
-
-			if ( $post && ! \Distributor\Utils\is_using_gutenberg( $post ) ) {
-				add_action( 'do_meta_boxes', __NAMESPACE__ . '\replace_revisions_meta_box', 10, 3 );
-				add_action( 'add_meta_boxes', __NAMESPACE__ . '\add_revisions_meta_box' );
-			}
+			add_action( 'do_meta_boxes', __NAMESPACE__ . '\replace_revisions_meta_box', 10, 3 );
+			add_action( 'add_meta_boxes', __NAMESPACE__ . '\add_revisions_meta_box' );
```
Comment (peterwilsoncc) [inline includes/syndicated-post-ui.php:32]: The `is_using_gutenberg()` function call is moved to the `do_meta_boxes` hook where it is used. This ensures all the editor functions are defined before attempting to use them. Specifically, in this instance, `use_block_editor_for_post_type()`.

**Review by iamdharmesh — COMMENTED** [review]
Thanks a lot for the great work here @peterwilsoncc, Code looks good to me just want you to check on the notes below.

- I am getting 404 for admin.min.css
image

- I think we should also add minimum requirements in the plugin header of the plugin’s main PHP file as well. Since WordPress 5.8 plugin [readme files are not parsed for requirements](https://core.trac.wordpress.org/ticket/48520). This means that headers `Requires PHP` and `Requires at least` are going to be parsed from plugin’s main PHP file.

Thanks

**Comment by peterwilsoncc** [comment]
@iamdharmesh I've added the requirements to the main plugin file in c1c93fb21b01a57cff5c7d39dc7ac10f360dc732 

You'll need to run `npm i; npm run build` in your command line for the css files to be generated.

**Comment by iamdharmesh** [comment]
Thanks for adding requirements to the main plugin file @peterwilsoncc.

> You'll need to run `npm i; npm run build` in your command line for the css files to be generated.

Yes, I am already doing this, but it seems the URL to load CSS file is wrong. it is `.../distributor/includes/dist/css/...` instead of `.../distributor/dist/css/...`.

Thanks

File: includes/bootstrap.php [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+		$asset_data = array(
+			'version'      => DT_VERSION,
+			'dependencies' => array(),
+		);
+		if ( file_exists( $asset_file ) ) {
+			$asset_data = require $asset_file;
+		}
+
+		// Dependencies only apply to JavaScript, not CSS files.
+		wp_enqueue_style( 'dt-admin-settings', plugins_url( '/dist/css/admin.min.css', __FILE__ ), array(), $asset_data['version'] );
```
Comment (iamdharmesh) [inline includes/bootstrap.php]:
```suggestion
		wp_enqueue_style( 'dt-admin-settings', plugins_url( '/dist/css/admin.min.css', __DIR__ ), array(), $asset_data['version'] );
```

**Comment by peterwilsoncc** [comment]
> it is `.../distributor/includes/dist/css/...` instead of `.../distributor/dist/css/...`.

Sorry, I missed that in the screen shot.

I've merged in your chagne

**Review by iamdharmesh — DISMISSED** [review]
Thanks for the changes @peterwilsoncc. All looks good to me now. :rocket: 

Thanks for your amazing work here.

File: README.md [hunk 4]
```diff
@@ -169,9 +169,15 @@ Enabling this will also provide more debugging information in your error log for
 
 ### Application Passwords and WordPress 5.6
 
-In WordPress 5.6, Application Passwords was merged into WordPress core with some limitations. From 5.6, Application Passwords is enabled by default only for live sites with HTTPS. To enable Application Passwords for development sites, you will need the following snippet:
+Application passwords are only available for live sites running over an HTTPS connection.
+
+For your local development environment will need these snippets to enable application passwords without the need for an HTTPS connection.  A local development environment is one that "can reach the internet but **is not reachable from the internet**".
```
Comment (dkotter) [inline README.md]:
```suggestion
For your local development environment, you will need these snippets to enable application passwords without the need for an HTTPS connection.  A local development environment is one that "can reach the internet but **is not reachable from the internet**".
```

**Review by dkotter — DISMISSED** [review]

**Review by dkotter — APPROVED** [review]

**Pull request merged**